### ISSUE TITLE
free function sweep BRepOffsetAPI_MakePipeShell  option to SetMode

### ISSUE THREAD

**Issue by lorenzncode** [CadQuery/cadquery#1639]
I did not expect this result from the sweep.

```py
import cadquery as cq
from cadquery.occ_impl.shapes import *

path = cq.Wire.makeHelix(2, 6, 4)

e1 = segment((4, 0, 0), (4, 0, 0.5))

res = sweep(e1, path)
```

![image](https://github.com/user-attachments/assets/cf4870ce-9025-4363-bddf-70f1887b0dd3)

I got the expected result after changing the sweep function to call `builder.SetMode(True)` here:
https://github.com/CadQuery/cadquery/blob/3cd327a2dbaf3bae6791463a3dcb0ad7a714b607/cadquery/occ_impl/shapes.py#L4956

![image](https://github.com/user-attachments/assets/754b87bc-659b-446a-8fbc-91f6cac6b71e)

Perhaps this can be set by default, or optionally as a function argument.

**Comment by lorenzncode** [comment]
If I understand the new sweep `aux` param usage correctly, this is supported now in #1692.

```py
import cadquery as cq
from cadquery.occ_impl.shapes import *

path = cq.Wire.makeHelix(2, 6, 4)

e1 = segment((4, 0, 0), (4, 0, 0.5))

res = sweep(e1, path, segment((0, 0, 0), (0, 0, path.endPoint().z)))
```

**Comment by adam-urbanczyk** [comment]
Solved by #1692
